### PR TITLE
Actions: CL Generator script - Adds documentation to the script and improves the push method

### PR DESCRIPTION
**Pull request by Crossedfall**
## About The Pull Request

Now that we know everything works, I've gone through and documented the script and improved the method that it uses to push the changes. This should resolve the issue where it wasn't including the relevant PR number in the commits.

## Changelog
:cl:
tweak: Further tweaks to the automated changelog generator
/:cl:

### CODE REVIEW

**Comment by Crossedfall** [comment]
As a note: Since the pushing is handled by the script now, there's no harm in exiting 'successfully' if that's the preference.

File: tools/changelog/generate_cl.py [hunk 1]
```diff
@@ -26,7 +46,7 @@
 
 if not pr_list.totalCount:
     print("Direct commit detected")
-    exit(1)
+    exit(1) # Change to '0' if you do not want the action to count this is a failure
```
Comment (qwertyquerty) [inline tools/changelog/generate_cl.py]: This *as* a failure. Might also help to say what "this" is

File: tools/changelog/generate_cl.py [hunk 2]
```diff
@@ -40,7 +60,7 @@
     cl_list = CL_SPLIT.findall(cl.group(2))
 except AttributeError:
     print("No CL found!")
-    exit(1)
+    exit(1) # Change to '0' if you do not want the action to count this is a failure
```
Comment (qwertyquerty) [inline tools/changelog/generate_cl.py]: This *as* a failure. Might also help to say what "this" is

File: tools/changelog/generate_cl.py [hunk 3]
```diff
@@ -56,18 +76,21 @@
 write_cl['changes'] = []
 
 for k, v in cl_list:
-    if k in tags['tags'].keys():
+    if k in tags['tags'].keys(): # Check to see if there are any of the correct tags
```
Comment (qwertyquerty) [inline tools/changelog/generate_cl.py]: Check to see if there are any valid tags?

File: tools/changelog/generate_cl.py [hunk 4]
```diff
@@ -56,18 +76,21 @@
...
+        yaml.dump(write_cl, cl_contents)
+        cl_contents.seek(0)
 
+        #Push the newly generated changelog to the master branch so that it can be compiled
+        repo.create_file(f"html/changelogs/AutoChangeLog-pr-{pr_number}.yml", f"Automatic changelog generation for PR #{pr_number} [ci skip]", content=f'{cl_contents.read()}', branch='master', committer=InputGitAuthor(git_name, git_email))
     print("Done!")
 else:
     print("No CL changes detected!")
-    exit(0)
+    exit(0) # Change to a '1' if you want the action to count this as a failure
```
Comment (qwertyquerty) [inline tools/changelog/generate_cl.py]: Might help to say what "this" is

**Pull request merged**